src/views/DataExploration/components/PageContent/tabs/Participants/index.tsx

```diff
@@ -468,18 +467,28 @@ const ParticipantsTab = ({ results, setQueryConfig, queryConfig, sqon }: OwnProp
   const { activeQuery } = useQueryBuilderState(DATA_EXPLORATION_QB_ID);
   const [selectedAllResults, setSelectedAllResults] = useState(false);
   const [selectedKeys, setSelectedKeys] = useState<string[]>([]);
+  const [selectedRows, setSelectedRows] = useState<IParticipantEntity[]>([]);
 
   useEffect(() => {
     if (selectedKeys.length) {
       setSelectedKeys([]);
+      setSelectedRows([]);
     }
     // eslint-disable-next-line
   }, [JSON.stringify(activeQuery)]);
 
   const getCurrentSqon = (): any =>
     selectedAllResults || !selectedKeys.length
       ? sqon
-      : generateSelectionSqon(TAB_IDS.PARTICIPANTS, selectedKeys);
+      : generateQuery({
+          newFilters: [
+            generateValueFilter({
+              field: PARTICIPANTS_SAVED_SETS_FIELD,
+              index: INDEXES.PARTICIPANT,
+              value: selectedRows.map((row) => row[PARTICIPANTS_SAVED_SETS_FIELD]),
+            }),
+          ],
+        });
```

Comment:
This piece of code seems to be repeated with slight variation a few times.

Reply:
J'y avais pensé mais j'ai l'impression qu'on se tire tout autant dans le pied en étant trop générique que pas assez. La dernière fonction qui était utiliser avant (generateSelectionSqon) est devenu un four tout avec des switch. Si on exporte le composant niveau ferlab plus tard, il y aura un intérêt à founir un fonction à importer mais pour le moment, je préfère ne pas mettre de fonction pour éviter de devoir la retirer tout de suite après car on a un nouveau corner-case. À voir une fois qu'on aura intégrer tous les tableaux si c'est reellement redondant ou non.